File: src/main/java/org/logister/android/LogisterTransport.java
package org.logister.android;

import org.json.JSONObject;

/** Transport used by LogisterClient. Tests and apps can provide their own implementation. */
public interface LogisterTransport {
    LogisterResponse send(
            String endpoint,
            String apiKey,
            JSONObject envelope,
            int connectTimeoutMs,
            int readTimeoutMs
    ) throws Exception;
}

File: src/main/kotlin/org/logister/android/LogisterTransport.kt
package org.logister.android

import org.json.JSONObject

/** Transport used by LogisterClient. Tests and apps can provide their own implementation. */
public fun interface LogisterTransport {
    @Throws(Exception::class)
    public fun send(
        endpoint: String,
        mobileIngestToken: String,
        envelope: JSONObject,
        connectTimeoutMs: Int,
        readTimeoutMs: Int
    ): LogisterResponse
}
